### PR TITLE
[BugFix] Validate kafka offset either by custom offset or kafka meta before alter

### LINKED ISSUE #13048: `FE` exit after `PAUSE` routine load task.
### Steps to reproduce the behavior (Required)
1. create table.
2. create routine load task and `pause` routine load before data loaded.
3. restart fe checkpoint.
### Expected behavior (Required)
restart successfully.
### Real behavior (Required)
```
2022-11-07 11:18:15,662 ERROR (stateListener|71) [EditLog.loadJournal():828] Operation Type 111
com.starrocks.common.DdlException: The specified partition 0 is not in the consumed partitions
        at com.starrocks.load.routineload.KafkaProgress.modifyOffset(KafkaProgress.java:139) ~[starrocks-fe.jar:?]
        at com.starrocks.load.routineload.KafkaRoutineLoadJob.modifyDataSourceProperties(KafkaRoutineLoadJob.java:530) ~[starrocks-fe.jar:?]
        at com.starrocks.load.routineload.RoutineLoadJob.modifyJob(RoutineLoadJob.java:1510) ~[starrocks-fe.jar:?]
        at com.starrocks.load.routineload.RoutineLoadManager.replayAlterRoutineLoadJob(RoutineLoadManager.java:620) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:790) [starrocks-fe.jar:?]
        at com.starrocks.catalog.Catalog.replayJournal(Catalog.java:2537) [starrocks-fe.jar:?]
        at com.starrocks.catalog.Catalog.transferToMaster(Catalog.java:1257) [starrocks-fe.jar:?]
        at com.starrocks.catalog.Catalog.access$900(Catalog.java:289) [starrocks-fe.jar:?]
        at com.starrocks.catalog.Catalog$4.runOneCycle(Catalog.java:2454) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:119) [starrocks-fe.jar:?]
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 87fbc7d

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13048

## Problem Summary(Required) ：

Altering the Kafka offset of a routine load job may cause FE to crash in various ways. I myself have committed two PRs to fix two of the scenarios.
https://github.com/StarRocks/starrocks/pull/8290
https://github.com/StarRocks/starrocks/pull/12227

The main reason is because, the Kafka partition offset from the job's metadata, won't initialize unless the job is scheduled, and it won't persist the progress unless data is loaded. Yet we often need it to validate the partitions from the statement. Thus in some cases, the Kafka offset between the leader and the followers may end up different.

The easiest way to fix all these meta-insistency problems is to initialize the offset and persist them in the first journal of creating this job. But if a new partition is created, the offset of this new partition still won't persist until data was loaded in it.

Another way to fix it is to avoid using the Kafka partition offset from the job's meta which won't persist in certain circumstances.

* If `customKafkaParition` is specified, then this job is created with specific partitions set by the user, which means only these partitions can be modified
* Otherwise, we shall check if the partition is validated by actually reading the Kafka meta from the Kafka proxy.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 6 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|3|6|50.00%|[398, 399, 400]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 14 (28.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|4|14|28.57%|[531, 532, 533, 534, 535, 537, 541, 542, 543, 544]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by gengjun-git — DISMISSED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java [hunk 1]
```diff
@@ -515,6 +518,35 @@ public void readFields(DataInput in) throws IOException {
...
+    /**
+     * add extra parameter check for changing kafka offset
+     * 1. if customKafkaParition is specified, only the specific partitions can be modified
+     * 2. otherwise, will check if partition is validated by actually reading kafka meta from kafka proxy
+     */
+    @Override
+    public void modifyJob(RoutineLoadDesc routineLoadDesc, Map<String, String> jobProperties,
+                          RoutineLoadDataSourceProperties dataSourceProperties, OriginStatement originStatement,
+                          boolean isReplay) throws DdlException {
+        if (!isReplay && dataSourceProperties != null && dataSourceProperties.hasAnalyzedProperties()) {
```
Comment (rickif) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java:530]: What will happen if `dataSourceProperties.hasAnalyzedProperties() == false`?

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Review by rickif — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 14 (28.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|4|14|28.57%|[531, 532, 533, 534, 535, 537, 541, 542, 543, 544]|

**Review by gengjun-git — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-2.4 branch-2.3 branch-2.2

**Pull request merged**